src/Bicep.Core/Syntax/TestDeclarationSyntax.cs

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Bicep.Core.Navigation;
+using Bicep.Core.Parsing;
+
+namespace Bicep.Core.Syntax
+{
+    public class TestDeclarationSyntax : StatementSyntax, ITopLevelNamedDeclarationSyntax
+    {
+        public TestDeclarationSyntax(IEnumerable<SyntaxBase> leadingNodes, Token keyword, IdentifierSyntax name, SyntaxBase path, SyntaxBase assignment, SyntaxBase value)
+            : base(leadingNodes)
+        {
+            AssertKeyword(keyword, nameof(keyword), LanguageConstants.TestKeyword);
+            AssertSyntaxType(name, nameof(name), typeof(IdentifierSyntax));
```

Comment:
nit: No need to assert the type of `name` here.

Reply:
Should I changed here? 
Or will this new PR include the changes to this line?

Reply:
Feel free to change this file since it's a new file.